test/jdk/sun/reflect/ReflectionFactory/ReflectionFactoryTest.java

```diff
@@ -328,6 +335,646 @@ static void newOptionalDataException() {
 
     }
 
+    private static final String[] names = {
+        "boolean_",
+        "final_boolean",
+        "byte_",
+        "final_byte",
+        "char_",
+        "final_char",
+        "short_",
+        "final_short",
+        "int_",
+        "final_int",
+        "long_",
+        "final_long",
+        "float_",
+        "final_float",
+        "double_",
+        "final_double",
+        "str",
+        "final_str",
+        "writeFields",
+    };
+
+    // test that the generated read/write objects are working properly
+    @Test
+    static void testDefaultReadWriteObject() throws Throwable {
+        Ser2 ser = new Ser2((byte) 0x33, (short) 0x2244, (char) 0x5342, 0x05382716, 0xf035a73b09113bacL, 1234f, 3456.0, true, new Ser3(0x004917aa));
+        ser.byte_ = (byte) 0x44;
+        ser.short_ = (short) 0x3355;
+        ser.char_ = (char) 0x6593;
+        ser.int_ = 0x4928a299;
+        ser.long_ = 0x24aa19883f4b9138L;
+        ser.float_ = 4321f;
+        ser.double_ = 6543.0;
+        ser.boolean_ = false;
+        ser.ser = new Ser3(0x70b030a0);
+        // first, ensure that each field gets written
+        MethodHandle writeObject = factory.defaultWriteObjectForSerialization(Ser2.class);
+        Assert.assertNotNull(writeObject, "writeObject not created");
+        boolean[] called = new boolean[19];
+        @SuppressWarnings("removal")
+        ObjectOutputStream oos = AccessController.doPrivileged((PrivilegedExceptionAction<ObjectOutputStream>) () -> new ObjectOutputStream() {
+            protected void writeObjectOverride(final Object obj) throws IOException {
```

Comment:
So far I'm unit testing each method to show that it works to its own specification; in order to fully integrate with multiple classes and nested objects, it seems as though I'd have to implement an entire custom serialization framework for the test, which does not seem feasible; the generated method does not relate in any way to nested serializations (that would be solely the job of the framework which uses it).

The only difference I can see between directly using the generated `default(Read|Write)Object` and one which uses them indirectly by way of a `Ser#(read|write)Object()` calling through `Object*Stream.default(Read|Write)Object()` is the call path through the test classes and mocks. But, I could possibly add a couple more tests with another pair of slightly different mocked streams which "tracks" the "current" object (of which there would just be one) and makes sure that `default(Read|Write)Object` also calls through the handle, just to show that it works, though I don't think this adds any surety or even code coverage to the overall tests since all of the additional code would exist only in the unit test (i.e. I'd be writing a test that proves that the test works, not that the feature works).

I do believe that this feature is fully covered by the existing tests, but if I am missing some subtle detail we can discuss further of course.